### ISSUE TITLE
Please consider allowing immutable config

### ISSUE THREAD

**Issue by iMartyn** [Chocobozzz/PeerTube#3754]
**Describe the problem to be solved**

Running peertube in many situations, it is useful for the config to be immutable.  This means read-only.

For example, it's not possible to sensibly package peertube as a helm chart for kubernetes because configmaps are read-only to the OS.

The same would be true of running peertube in Heroku and OpenShift.

**Describe the solution you would like:**

Simple: do not fail to run if the configuration is readonly.  This is a choice of the administrator of the system, and can be a security concern.  If you refuse to see the advantage of allowing immutable configuration by default then at least allow it to be turned from an error into a warning by a flag or environment variable.

**Describe alternatives you have considered**  

The only solution at present is to copy the config to somewhere writable on startup which is a messy and error-prone solution.

**Comment by Chocobozzz** [comment]
Hello,

I'm not sure to understand: what do you mean by read-only? readonly permission? readonly filesystem?

What is the current situation? PeerTube crashes?

**Comment by iMartyn** [comment]
Yes, readonly filesystem.  Same is true of readonly permission afaik because of the "check I can write to the config after upgrade to change it" part of boot.

Yes, last time I ran peertube in kubernetes with a config mounted from a ConfigMap, peertube errored out and refused to start.

When I raised a bug (#1994) for this, it was considered "correct" that a readonly config so I am now raising a feature request to change the behavior.  Perhaps the feature-request is better written and better understood.

**Comment by Chocobozzz** [comment]
Oh okay you're talking about the Docker container that copies the config at startup. I think we can accept a PR that adds an env check to avoid config copy. 

But I'm not sure to understand why you don't build the docker image by yourself, so you can configure/update the entrypoint depending on your needs. We provide a docker image for our docker-compose setup, it's not designed to be used on kubertenes or other tools we (peertube maintainers) don't really know.

**Comment by iMartyn** [comment]
Building a container image yourself is something that shouldn't be necessary, the official docker container shouldn't make such assumptions that it is only for use in the very specific docker compose setup preferred by the peertube maintainers.  If it is only for that it should be labelled very clearly.

Building a container yourself requires a whole CI setup to make sure you get security updates etc.  An official docker container should always be used where possible.

But I think my point still stands, if the peertube official docker container refuses to start with a readonly config file, it refuses to start with a readonly config file.  It shouldn't make the assumption that the file is writable, as immutable config is a good thing, and a secure thing.

**Comment by pingiun** [comment]
Hey I wanna work on this, I have an idea on how it could work but I'll explain that in the PR I'm gonna open soon
